### PR TITLE
feat(route): add 北京师范大学-香港浸会大学联合国际学院 UIC

### PR DESCRIPTION
**Pull request by heimoshuiyu**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/uic/news/top_stories
/uic/news/in_brief
/uic/news/features
/uic/news/media
/uic/lrc
/uic/fst
/uic/career/ygtz
/uic/career/xsfx
/uic/career/gg
/uic/career/xngs
/uic/career/msjq
/uic/career/sjzxx
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [x] `Puppeteer`

## Note / 说明

这b学校的网站证书有问题（证书链不完整）导致 Linux 下的部分软件 curl / nodejs fetch / python requests 等无法验证其证书。

<https://myssl.com/uic.edu.cn?domain=uic.edu.cn&status=q>

需要设置 [NODE_TLS_REJECT_UNAUTHORIZED = '0'](https://github.com/DIYgod/RSSHub/commit/741a3b8a8e0e2fa8f347a0443e417170ebf54e06) 环境变量

### CODE REVIEW

File: lib/routes/uic/namespace.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import type { Namespace } from '@/types';
+
+export const namespace: Namespace = {
+    name: 'BNU-HKBU United International College',
```
Comment (TonyRL) [inline lib/routes/uic/namespace.ts:4]:
```suggestion
    name: 'Beijing Normal University-Hong Kong Baptist University United International College',
```

File: lib/routes/uic/career.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import puppeteer from '@/utils/puppeteer';
+import * as cheerio from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/career/:tag',
+    categories: ['university'],
+    example: '/uic/career/ygtz',
+    parameters: {
+        tag: 'tag, ygtz, xsfx, gg, xngs, msjq, sjzxx',
```
Comment (TonyRL) [inline lib/routes/uic/career.ts:11]: Please provide more concise explanations.

File: lib/routes/uic/career.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    features: {},
```
Comment (TonyRL) [inline lib/routes/uic/career.ts:13]:
```suggestion
```

File: lib/routes/uic/career.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    name: 'UIC Career',
```
Comment (TonyRL) [inline lib/routes/uic/career.ts:14]:
```suggestion
    name: 'Career',
```

File: lib/routes/uic/career.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            request.abort();
+        } else {
+            request.continue();
+        }
+    });
+    await page.goto(rootUrl, {
+        waitUntil: 'networkidle2',
+    });
+    const response = await page.content();
+    await page.close();
```
Comment (TonyRL) [inline lib/routes/uic/career.ts:37]: No need to use puppetter.
You can extract the compressed content using regex
 https://github.com/DIYgod/RSSHub/blob/379aa7d98f95820e9498b8671bac865df0b1788c/lib/routes/csu/career.ts#L56-L62

File: lib/routes/uic/fst.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import { Route } from '@/types';
+import got from '@/utils/got';
+import * as cheerio from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/fst',
+    categories: ['university'],
+    example: '/uic/fst',
+    name: 'UIC FST',
```
Comment (TonyRL) [inline lib/routes/uic/fst.ts:10]:
```suggestion
    name: 'FST',
```

File: lib/routes/uic/fst.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    maintainers: ['heimoshuiyu'],
+    features: {},
```
Comment (TonyRL) [inline lib/routes/uic/fst.ts:12]:
```suggestion
```

File: lib/routes/uic/lrc.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import { Route } from '@/types';
+import got from '@/utils/got';
+import * as cheerio from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/lrc',
+    categories: ['university'],
+    example: '/uic/lrc',
+    name: 'UIC LRC',
```
Comment (TonyRL) [inline lib/routes/uic/lrc.ts:10]:
```suggestion
    name: 'LRC',
```

File: lib/routes/uic/lrc.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    maintainers: ['heimoshuiyu'],
+    features: {},
```
Comment (TonyRL) [inline lib/routes/uic/lrc.ts:12]:
```suggestion
```

File: lib/routes/uic/fst.ts [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+            .map(async (_, item) => {
+                const link = $(item).attr('href');
+                const subResponse = await got.get('https://fst.uic.edu.cn/news_events/' + link);
+                const $$ = cheerio.load(subResponse.data);
+                return {
+                    title: $(item).find('h3').text(),
+                    link,
+                    pubDate: parseDate($(item).find('div.date').text(), 'YYYY/MM/DD'),
+                    description: $$('div.v_news_content').html() || '',
+                };
```
Comment (TonyRL) [inline lib/routes/uic/fst.ts:34]: Please cache the returned object

File: lib/routes/uic/lrc.ts [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+            .map(async (_, item) => {
+                const link = $(item).find('a').attr('href');
+                const subResponse = await got.get('https://lrc.uic.edu.cn/index/' + link);
+                const $$ = cheerio.load(subResponse.data);
+                return {
+                    title: $(item).find('td.list-title').text(),
+                    link,
+                    pubDate: parseDate($(item).find('td.list-date').text(), 'YYYY年MM月DD日'),
+                    description: $$('div.v_news_content').html() || '',
+                };
```
Comment (TonyRL) [inline lib/routes/uic/lrc.ts:34]: Please cache the returned object

File: lib/routes/uic/news.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
+import { Route } from '@/types';
+import got from '@/utils/got';
+import * as cheerio from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/news/:sub',
+    parameters: {
+        sub: 'sub',
```
Comment (TonyRL) [inline lib/routes/uic/news.ts:9]: Please provide more concise explanations.

File: lib/routes/uic/news.ts [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+    },
+    categories: ['university'],
+    example: '/uic/news/xw',
+    name: 'UIC News',
```
Comment (TonyRL) [inline lib/routes/uic/news.ts:13]:
```suggestion
    name: 'News',
```

File: lib/routes/uic/news.ts [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    maintainers: ['heimoshuiyu'],
+    features: {},
```
Comment (TonyRL) [inline lib/routes/uic/news.ts:15]:
```suggestion
```

File: lib/routes/uic/news.ts [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+                const $$ = cheerio.load(page.data);
+                const description = $$('div.v_news_content').html();
+                const dateText = $$('span.nry_data').text().trim();
+                const pubDate = parseDate(dateText.split('：')[1], 'YYYY-MM-DD');
+                return {
+                    title: $(item).find('a').text(),
+                    link,
+                    description: description || '',
+                    pubDate,
+                };
```
Comment (TonyRL) [inline lib/routes/uic/news.ts:39]: Please cache the returned object

File: website/docs/routes/university.mdx [hunk 16]
```diff
@@ -185,6 +185,40 @@
...
+<Route author="heimoshuiyu" example="/uic/fst" />
+
+### UIC 学生职业发展中心 {#bnu-hkbu-uic-career}
+
+<Route author="heimoshuiyu" example="/uic/career/ygtz" path="/uic/career/:tag" paramsDesc={['新闻子类']}>
+  |  ygtz  |  xsfx   |   gg    |    xngs    |   msjq  |  sjzxx |
+  | -----  |  -----  |  ----   |  --------- | -----  |  ------ |
+  | 中心新闻 | 学生风采 | 政府公告 | 就业质量报告 | 就业指南 | 升学讲座 |
+</Route>
+
```
Comment (TonyRL) [inline website/docs/routes/university.mdx:221]: Do not edit the docs directly. All manual changes will be overwritten by each route.

**Comment by heimoshuiyu** [comment]
For some reasons, I will not continue to handle this PR. If someone is willing to take over, please feel free to use my code

**Pull request closed without merging**